notebooks/03-query/07-functions-datetime.ipynb

```diff
@@ -0,0 +1,963 @@
+    "Using the `WHERE` condition for filters, the next cell uses the `TIME_IN_INTERVAL` to generate a 15-minutely breakdown, per continent, of the number of sessions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "48b702cd-9d38-45fa-a67a-dc03d8fd1883",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT DISTINCT  \n",
+    "  \"continent\",\n",
+    "  COUNT(DISTINCT \"session\") FILTER (WHERE TIME_IN_INTERVAL(__time, '2019-08-25T03:00:00/PT15M')) AS \"sessions_0\",\n",
+    "  COUNT(DISTINCT \"session\") FILTER (WHERE TIME_IN_INTERVAL(__time, '2019-08-25T03:15:00/PT15M')) AS \"sessions_1\",\n",
+    "  COUNT(DISTINCT \"session\") FILTER (WHERE TIME_IN_INTERVAL(__time, '2019-08-25T03:30:00/PT15M')) AS \"sessions_2\",\n",
+    "  COUNT(DISTINCT \"session\") FILTER (WHERE TIME_IN_INTERVAL(__time, '2019-08-25T03:45:00/PT15M')) AS \"sessions_3\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "WHERE TIME_IN_INTERVAL(__time, '2019-08-25T03/PT1H')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6114423e-1090-4d9a-b973-bbb55f216a38",
+   "metadata": {},
+   "source": [
+    "Run the following cell to see an example of a single query that compares data from two time periods using a `JOIN`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4ea76b33-8a04-4fff-b0e5-ae4688b019f8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "WITH recent_sessions AS\n",
+    "(\n",
+    "    SELECT \"continent\", COUNT(DISTINCT \"session\") AS \"sessions\"\n",
+    "    FROM \"example-koalas-fndatetime\"\n",
+    "    WHERE TIME_IN_INTERVAL(__time, '2019-08-25T06/PT1H')\n",
+    "    GROUP BY 1\n",
+    "),\n",
+    "older_sessions_15minutely AS\n",
+    "(\n",
+    "    SELECT TIME_FLOOR(__time,'PT15M'), \"continent\", COUNT(DISTINCT \"session\") AS \"sessions\"\n",
+    "    FROM \"example-koalas-fndatetime\"\n",
+    "    WHERE TIME_IN_INTERVAL(__time, '2019-08-25T00/PT12H')\n",
+    "    GROUP BY 1, 2\n",
+    "),\n",
+    "older_sessions AS\n",
+    "(\n",
+    "    SELECT \"continent\", AVG(\"sessions\") AS \"sessions\"\n",
+    "    FROM older_sessions_15minutely\n",
+    "    GROUP BY 1\n",
+    ")\n",
+    "\n",
+    "SELECT\n",
+    "    recent_sessions.\"continent\",\n",
+    "    recent_sessions.sessions AS \"recent\",\n",
+    "    older_sessions.sessions AS \"historical_average\"\n",
+    "FROM recent_sessions\n",
+    "LEFT JOIN older_sessions ON recent_sessions.continent = older_sessions.continent\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "625b2d58-42e0-4ae7-bfd9-6f1d8eccf2ad",
+   "metadata": {},
+   "source": [
+    "The `GROUP BY` function can be used at ingestion time (also known as rollup in JSON-based ingestion) to pre-calculate commonly used aggregations.\n",
```

Comment:
Should we call it "JSON-based ingestion" or just Native Ingestion?

Reply:
Good spot. I checked the docs and it's "Classic batch" and "streaming" are separated, with "JSON-based" being used specifically to refer to batch ingestion. I've therefore just updated this notebook with "classic ingestion" throughout, and mentioned both batch and streaming.